include/xtensor/xoperation.hpp

```diff
@@ -959,6 +959,33 @@ namespace xt
     {
         return detail::make_xfunction<typename detail::cast<R>::functor>(std::forward<E>(e));
     }
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param e an \ref xexpression
+    * @return a boolean array
+    */
+    template <class E, class T>
+    inline auto isin(E&& e, std::initializer_list<T> values) noexcept
+        -> detail::xfunction_type_t<detail::equal_to, E, T>
+    {
+        using return_type = detail::xfunction_type_t<detail::equal_to, E, T>;
+
+        auto iter = values.begin();
+        return_type b = xt::equal(e, *iter);
+        iter++;
+
+        for (; iter != values.end(); iter++) {
+            b += xt::equal(e, *iter);
+        }
+
+        return b;
```

Comment:
`b` is an instance of `xfunction` which is not assignable (since it is a non-evaluated expression). Since the number of elements in values is not known at compile time, I'm not sure you can implement an unevaluated function for `isin` this way. Besides, this broadcast each value of `iter` agains `e` for each entry in `e`. Therefore, if you need to access a single value of the result, you will still compute the whole expression.

For returning an unevaluated expression, I think that a new functor must be created, which captures the `values` argument. A `lambda` is a perfect candidate for that, and thanks to `make_lambda_xfunction`, turning it into a new xfunction is a one-liner. So the implementation would look something like:

```cpp
template <class E, class T>
inline auto isin(E&& e, std::initializer_list<T> values) noexcept
{
    auto lambda = [&values](const auto& t)  { return std::find(values.begin(), values.end(), t) != values.end(); }
    return make_lambda_xfunction(lambda, std::forward<E>(e));
}
```

An overload accepting a pair of iterators can also be provided, so that you can easily use `isin` with any kind of containers. What do you think?

Reply:
Nice! While trying I got 
```none
In file included from /Users/tdegeus/miniconda3/envs/test/include/xtensor/xtensor.hpp:20:
In file included from /Users/tdegeus/miniconda3/envs/test/include/xtensor/xcontainer.hpp:24:
/Users/tdegeus/miniconda3/envs/test/include/xtensor/xmath.hpp:1046:19: error: non-const lvalue reference to type '(lambda at
      /Users/tdegeus/miniconda3/envs/test/include/xtensor/xoperation.hpp:975:18)' cannot bind to a temporary of type 'typename
      remove_reference<(lambda at /Users/tdegeus/miniconda3/envs/test/include/xtensor/xoperation.hpp:975:18) &>::type' (aka '(lambda at
      /Users/tdegeus/miniconda3/envs/test/include/xtensor/xoperation.hpp:975:18)')
                : m_lambda(std::move(lmbd))
                  ^        ~~~~~~~~~~~~~~~
/Users/tdegeus/miniconda3/envs/test/include/xtensor/xmath.hpp:1096:31: note: in instantiation of member function
      'xt::detail::lambda_adapt<(lambda at /Users/tdegeus/miniconda3/envs/test/include/xtensor/xoperation.hpp:975:18) &>::lambda_adapt'
      requested here
        return xfunction_type(detail::lambda_adapt<F>(std::forward<F>(lambda)), std::forward<E>(args)...);
                              ^
```

Reply:
Does it help if you std::move(lambda) ?

Reply:
Yes! But the results in only `true`, even when it shouldn't

Reply:
I will figure out tomorrow what you are doing ;)

Reply:
I'm impressed @JohanMabille !